scipy/stats/_stats.pyx

```diff
@@ -1,11 +1,10 @@
 from cpython cimport bool
 from libc cimport math
+from libc.math cimport NAN, INFINITY, M_PI as PI
```

Comment:
PR mostly looks good to me. Maybe slight concern about the CI failure in `stats` unless we can see them failing elsewhere, because of the `NAN` shift:

```
FAILED scipy/stats/tests/test_rank.py::TestRankData::test_nan_policy_omit_3d[min-0] - RuntimeWarning: invalid value encountered in cast
FAILED scipy/stats/tests/test_rank.py::TestRankData::test_nan_policy_omit_3d[max-0] - RuntimeWarning: invalid value encountered in cast
FAILED scipy/stats/tests/test_rank.py::TestRankData::test_nan_policy_omit_3d[dense-0] - RuntimeWarning: invalid value encountered in cast
FAILED scipy/stats/tests/test_rank.py::TestRankData::test_nan_policy_omit_3d[ordinal-0] - RuntimeWarning: invalid value encountered in cast
```

Maybe just an unfortunate coincidence, but not sure, can check locally later maybe.

Reply:
For the record, I couldn't reproduce on this branch locally on Ubuntu with Python 3.10 or 3.11.

Reply:
Another report of that at https://github.com/scipy/scipy/pull/17500#issuecomment-1328354630. It seems to have passed now - I think that test is flaky.